common/src/Renderer/EntityRenderer.cpp

```diff
@@ -252,7 +252,8 @@ namespace TrenchBroom {
                 const auto center = vm::vec3f(entity->bounds().center());
                 
                 const auto toCam = renderContext.camera().position() - center;
-                if (squaredLength(toCam) > maxDistance2) {
+                // only distance cull for perspective camera, since the 2D one is always very far from the level
+                if (renderContext.camera().perspectiveProjection() && squaredLength(toCam) > maxDistance2) {
```

Comment:
I wonder if it's not better to delegate this decision to the camera in some way, since now we're doing essentially doing a type check (in many places) to determine how some aspects should behave. Instead of this, we might as well add a virtual function

```
bool Camera::shouldDrawEntityIndicator(...)
```

that decides this. OTOH this overloads the camera class with a lot of responsibilities that it shouldn't be concerned with either.

I think we should leave it as it is for now, but I prefer to centralize these kinds of decisions in one place and solve them by inheritance and polymorphism rather than if / then / else all over the place.

Reply:
Yeah. We could make it a bit clearer why the code needs the type, something like `bool Camera::has3DPosition() const;` or `bool Camera::distanceCull(vm::vec3 point, float maxdist) const;` that would always return false for 2d cameras.

These type checks are pretty benign since we can easily do "find all references" on `perspectiveProjection`/`orthographicProjection` if we want to refactor them.